mongo_connector/oplog_manager.py

```diff
@@ -341,11 +358,25 @@ def filter_oplog_entry(self, entry):
         if not self._fields:
             return entry
 
-        def pop_excluded_fields(doc):
-            # always include _id
-            fields_with_id = self._fields.union(set(['_id']))
-            for key in set(doc) - fields_with_id:
```

Comment:
Here we're using field_with_id instead of fields because we want to make sure we're never removing _id from the fields that get copied over to the sink.